src/modules/accommodations/dto/requests/update-accommodation.req.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  IsBoolean,
+  IsDateString,
+  IsNumber,
+  IsOptional,
+  IsString,
+  Min,
+  ValidateNested,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+import { AccommodationImageDto } from './create-accommodation-image.req';
+import { AccommodationAmenityDto } from './create-accommodation-amenity.req';
+import { UpdateAccommodationAddressDto } from './update-accommodation-address.req';
+import { Type } from 'class-transformer';
+
+export class UpdateAccommodationDto {
+  @ApiProperty({ description: 'Name of the accommodation', required: false })
+  @IsOptional()
+  @IsString()
+  name?: string;
+
+  @ApiProperty({
+    description: 'Description of the accommodation',
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  description?: string;
+
+  @ApiProperty({
+    description: 'Cover image URL of the accommodation',
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  coverImage?: string | undefined;
+
+  @ApiProperty({
+    description: 'Price per night of the accommodation',
+    required: false,
+  })
+  @IsOptional()
+  @IsNumber()
+  price?: number;
```

Comment:
Please use undefined explicitly instead of ? optional operator. Make sure that all fields are updated accordingly